src/Components/ArticleCard/ArticleCard.jsx

```diff
@@ -8,12 +8,15 @@ const ArticleCard = ({ data }) => {
   const [bookmarked, setBookmark] = useState(data?.bookmarked);
   const handleBookmark = () => {
     /* Make requests to backend to update bookmark */
-    setBookmark(!bookmarked);
+    setBookmark((prevState) => !prevState);
   };
 
   return (
     <div className={styles.main}>
       <div className={styles.metadata}>
+        <Link to={data?.authorprofile}>
```

Comment:
Few changes on how you get the author data after creating the `author` property as suggested above.